lib/validator.js

```diff
@@ -115,6 +117,90 @@ const getValidator = (input) => {
   return validatorList[input];
 };
 
+class TypeError extends Error {
+  constructor(input) {
+    super(`input could not be coerced to type: ${input}`);
+  }
+}
+
+/**
+ * Cast to a string or throw an error.
+ * @param input
+ * @returns {string}
+ */
+const castToString = input => {
+  if(isArray(input) || isObject(input)) {
+    throw new TypeError('string');
+  }
+  else { return String(input); }
+};
+
+
+/**
+ * Cast to a string or throw an error.
+ * @param input
+ * @returns {string}
+ */
+const castToNumber = input => {
+  if(isNumber(input)) {
+    return input;
+  } else if(isString(input)) {
+    const temp = Number(input);
+    if(isNumber(temp)) {
+      return temp;
+    } else { throw new TypeError('number'); }
+  } else { throw new TypeError('number'); }
+};
+
+/**
+ * Cast to a string or throw an error.
+ * @param input
+ * @returns {string}
+ */
+const castToBoolean = input => {
+  if(isBoolean(input)) {
+    return input;
+  } 
+  else if(isString(input)) {
+    if(input.toLowerCase() === 'true') { return true; }
```

Comment:
Even though this totally works, I think you should aim to simplify logic like this in the future (instead of having super nested if/elses and calling the same error message multiple time)